src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpAddMissingUsingsOnPaste.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpAddMissingUsingsOnPaste : AbstractEditorTest
+    {
+        public CSharpAddMissingUsingsOnPaste(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory)
+        {
+        }
+
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.AddMissingImports)]
+        public void VerifyAddUsingsOnPaste()
+        {
+            var markup = @"
+using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+    }
+
+    $$
+}";
+            SetUpEditor(markup);
+
+            using (var telemetry = VisualStudio.EnableTestTelemetryChannel())
+            {
+                VisualStudio.Editor.Paste(@"
+Task DoThingAsync() => Task.CompletedTask;");
```

Comment:
I would not expect any using directives to be added in this case, because the code was not originally copied from source code inside Visual Studio.

Reply:
That was discussed as a design, but ultimately from #45853 we indicated that timeboxing was more appropriate. If the code is originally copied from Visual Studio we could speed up the process by knowing the usings without having to calculate. That work is not covered in this PR, and out of scope for #45853